backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java

```diff
@@ -97,35 +100,63 @@ private static void migrateProjects() {
+          Map<String, GetResourcesResponseItem> responseItemMap =
+              responseItems.stream()
+                  .collect(Collectors.toMap(GetResourcesResponseItem::getResourceId, item -> item));
+
           for (ProjectEntity project : projectEntities) {
-            WorkspaceDTO workspaceDTO =
-                roleService.getWorkspaceDTOByWorkspaceId(
-                    userInfoMap.get(project.getOwner()),
-                    project.getWorkspace(),
-                    project.getWorkspace_type());
-            // if projectVisibility is not equals to ResourceVisibility.ORG_SCOPED_PUBLIC then
-            // ignore the CollaboratorType
-            roleService.createWorkspacePermissions(
-                workspaceDTO.getWorkspaceName(),
-                project.getId(),
-                project.getName(),
-                Optional.of(Long.parseLong(project.getOwner())),
-                ModelDBServiceResourceTypes.PROJECT,
-                CollaboratorPermissions.newBuilder()
-                    .setCollaboratorType(CollaboratorTypeEnum.CollaboratorType.READ_ONLY)
-                    .build(),
-                getResourceVisibility(
-                    Optional.ofNullable(
-                        WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type())),
-                    VisibilityEnum.Visibility.forNumber(project.getProject_visibility())));
+            if (project.getOwner() != null && !project.getOwner().isEmpty()) {
+              WorkspaceDTO workspaceDTO =
+                  roleService.getWorkspaceDTOByWorkspaceId(
+                      userInfoMap.get(project.getOwner()),
+                      project.getWorkspace(),
+                      project.getWorkspace_type());
+              // if projectVisibility is not equals to ResourceVisibility.ORG_SCOPED_PUBLIC then
+              // ignore the CollaboratorType
+              roleService.createWorkspacePermissions(
+                  workspaceDTO.getWorkspaceName(),
+                  project.getId(),
+                  project.getName(),
+                  Optional.of(Long.parseLong(project.getOwner())),
+                  ModelDBServiceResourceTypes.PROJECT,
+                  CollaboratorPermissions.newBuilder()
+                      .setCollaboratorType(CollaboratorTypeEnum.CollaboratorType.READ_ONLY)
+                      .build(),
+                  getResourceVisibility(
+                      Optional.ofNullable(
+                          WorkspaceTypeEnum.WorkspaceType.forNumber(project.getWorkspace_type())),
+                      VisibilityEnum.Visibility.forNumber(project.getProject_visibility())));
+            } else {
+              GetResourcesResponseItem resourceDetails = responseItemMap.get(project.getId());
+              roleService.createWorkspacePermissions(
+                  resourceDetails.getWorkspaceId(),
+                  Optional.empty(),
```

Comment:
This argument is new compared to the baseline above. What is it? I tend to get confused with overloaded methods =)

Reply:
it is unused argument of `createWorkspacePermissions` which i will remove in follow up PR. baseline above method refactor already so it is not there. Sorry for confusion.